### PR TITLE
Fix: load empty collection, not object

### PR DESCRIPTION
**Pull request by Tilix4**
## Changelog Description
Make sure it applies to collections

### Third party change
- Fixed import

## Test
- Load `Auntie_ESTAB_001_A_JOUR` in prod test

### CODE REVIEW

File: openpype/hosts/blender/api/plugin.py [hunk 1]
```diff
@@ -766,7 +766,8 @@ def _load_library_datablocks(
                         col
                         for col in d.children_recursive
                         # Don't add empty collections
-                        if col.children or col.all_objects
+                        if isinstance(col, bpy.types.Collection)
```
Comment (kaamaurice) [inline openpype/hosts/blender/api/plugin.py]: Could be just simplified like this ?
```python
                if isinstance(d, bpy.types.Collection):
                    override_datablocks.update(
                        col
                        for col in d.children_recursive
                        # Don't add empty collections
                        if col.children or col.all_objects
                    )
                    override_datablocks.update(d.all_objects)
```

**Review by kaamaurice — CHANGES_REQUESTED** [review]
Should work fine but can be optimized.

File: openpype/hosts/blender/api/plugin.py (hunk 1 above)
Comment (Tilix4) [reply openpype/hosts/blender/api/plugin.py]: Indeed

Comment (Tilix4) [reply openpype/hosts/blender/api/plugin.py]: In fact no, we are getting it wrong. The idea is to collect all overridden datablocks of `d`, collections and objects. But we don't want to add empty collections, that's all. Therefore it must be done in another way

**Comment by Sharkitty** [comment]
Code looks good, but can you be more precise on what we're supposed to check in the testing note?

File: openpype/hosts/blender/api/capture.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 import bpy
 
 from .lib import maintained_time
-from .plugin import deselect_all, create_blender_context
+from .plugin import deselect_all
```
Comment (Sharkitty) [inline openpype/hosts/blender/api/capture.py:9]: `create_blender_context` is still referenced twice in this file, you shouldn't remove its import.

Comment (Sharkitty) [reply openpype/hosts/blender/api/capture.py:9]: Actually... It's not in `plugin.py`, but its references in `catpure.py` are problematic then

Comment (Tilix4) [reply openpype/hosts/blender/api/capture.py:9]: It doesn't exist in the `plugin.py` and it is never called with the current implementation. It is removed in later code. It is not clean I agree.

Comment (Tilix4) [reply openpype/hosts/blender/api/capture.py:9]: It'll be cleaned later on.

**Comment by Sharkitty** [comment]
The test passes, but it also passes without your fix, I'll try using an older version of the mentioned asset to see if I can get it to fail on a branch that does not contain this fix, then try again.

**Comment by Sharkitty** [comment]
> The test passes, but it also passes without your fix, I'll try using an older version of the mentioned asset to see if I can get it to fail on a branch that does not contain this fix, then try again.

I can't find a case in which this fix makes a difference sadly.

**Comment by Tilix4** [comment]
> > The test passes, but it also passes without your fix, I'll try using an older version of the mentioned asset to see if I can get it to fail on a branch that does not contain this fix, then try again.
> 
> I can't find a case in which this fix makes a difference sadly.

Indeed, I'm glad it works as usually, that was the goal. It would crash if an instance targets a single object and is loaded (I think). But the main test was to make sure it works as before.

**Review by Sharkitty — APPROVED** [review]

**Pull request merged**